scripts/policy/protocols/ssl/decryption.zeek

```diff
@@ -0,0 +1,106 @@
+#! Decrypt SSL/TLS payloads
+
+@load base/frameworks/input
+@load base/frameworks/notice
+@load base/protocols/conn
+@load base/protocols/ssl
+
+module SSL;
+
+# Local
+const input_stream_name = "input-tls-keylog-file";
+
+type Idx: record {
+	client_random: string;
+};
+
+type Val: record {
+	secret: string;
+};
+
+global randoms: table[string] of string = {};
+
+export {
+	redef record Info += {
+		# decryption uses client_random as identifier
+		client_random: string &log &optional;
+	};
+
+	const keylog_file = getenv("ZEEK_TLS_KEYLOG_FILE") &redef;
```

Comment:
just to check - did you ever end up implementing the keys back to Zeek via broker? Or did you only use the keylog file?

Edit: I see found the broker bit later. This probably should be docmented here.

The current approach of key ingestion also will not work in a cluster setting - is that something that you might be interesting of fixing?

Reply:
You are correct, currently the keys can either be received by broker or by ingesting a slightly modified keyfile on launch. Is there an equivalent of the Input Framework for cluster mode? Also what would it take to make key storage cluster compatible? Can we just store the secrets in the proxy?